lib/shrine/plugins/hanami.rb

```diff
@@ -3,65 +3,114 @@
 class Shrine
   module Plugins
     module Hanami
-      def self.configure(uploader, validations: nil)
-        uploader.opts[:hanami_validations] = validations
-      end
-
       module AttachmentMethods
         def initialize(name)
           super
 
-          if shrine_class.opts[:hanami_validations]
-            module_eval <<-RUBY, __FILE__, __LINE__ + 1
-              def validate
-                super
-                #{name}_attacher.errors.each do |message|
-                  errors.add(:#{name}, message)
-                end
+          module_eval <<-RUBY, __FILE__, __LINE__ + 1
+            module EntitySupport
+              attr_reader :attributes
+              def initialize(attributes)
+                attachment = attributes[:#{name}]
+                @_#{name} = attachment
+                self.#{name}_attacher
+                super(attributes)
               end
-            RUBY
-          end
+
+              def #{name}_data=(data)
```

Comment:
@nessur It will raise `RuntimeError: can't modify frozen XYZ`, just like when trying to assign any "normal" entity attribute. I think this behaviour makes sense.